Add `status` field to the plugin entry to show the maturity or release stage of the plugin

superstar54 opened the pull request:
This PR refactor the plugin manager:

- Add a PluginManager class, and import it inside the Jupyter notebook
- Add `status`, `category` and `requires_aiidalab_qe` fields.
- Add test for the plugin manager
- Read the `plugins.yaml` file from github instead of from local.
- Additionally, rename the `search_jobs` files to `calculation_history`


## Add `status` field


Each plugin has a **status badge** indicating its current development stage:  

| **Status**        | **Meaning** |
|-------------------|------------|
| 🟠 **Experimental** | The plugin is in an early stage of development and may be unstable. Expect rapid changes and potential issues. |
| 🟡 **Beta** | The plugin is functional and being tested, but some features may still be refined. |
| 🔵 **Stable** | The plugin is feature-complete, has undergone significant testing, and is considered reliable for most use cases. Minor updates or bug fixes may still occur. |
| 🟢 **Production** | The plugin is fully stable, widely tested, and officially recommended for all users. It follows strict maintenance and support policies. |

Here is an example of the UI:

![Screenshot from 2025-03-05 11-45-25](https://github.com/user-attachments/assets/c820d5f2-cc8f-4f1f-8fb0-384b71ea886f)

## Read the `plugins.yaml` file from github instead of from local.
In this way, a developer can continue adding new plugins after the app is released.

## Add `requires_aiidalab_qe` field
e.g., `requires_aiidalab_qe: '>=24.10.0'`
Disable the `install` button, if the aiidalab_qe is fulfilled.



## Add available plugins in step 2

![Screenshot from 2025-03-05 14-33-50](https://github.com/user-attachments/assets/ad9a99d8-242d-4a8e-b6e9-fefdf4177cf3)

superstar54 commented:
Hi @edan-bainglass , all tests passed, ready for review.

edan-bainglass commented on src/aiidalab_qe/app/configuration/__init__.py:
```diff
@@ -99,12 +102,31 @@ def _render(self):
         )
 
+        self.install_new_plugin_button = LinkButton(
+            description="Open Plugin Store",
```
```suggestion
            description="Plugin store",
```

edan-bainglass commented on src/aiidalab_qe/app/static/templates/plugin_manager.jinja:
```diff
@@ -0,0 +1,58 @@
+        You can install and uninstall plugins directly from this page.
+    </p>
+
+    <h3 style="margin-top: 20px;">Plugin Status Explanation</h3>
```
```suggestion
    <h3 style="margin-top: 20px;">Plugin status explanation</h3>
```

edan-bainglass commented on src/aiidalab_qe/app/static/templates/plugin_manager.jinja:
```diff
@@ -0,0 +1,58 @@
+        </tbody>
+    </table>
+
+    <h3 style="margin-top: 20px;">Available Plugins</h3>
```
```suggestion
    <h3 style="margin-top: 20px;">Available plugins</h3>
```

edan-bainglass commented on src/aiidalab_qe/app/static/templates/plugin_manager.jinja:
```diff
@@ -0,0 +1,58 @@
+            </tr>
+            <tr>
+                <td style="border: 1px solid #ddd; padding: 8px;">🟡 <strong>Beta</strong></td>
+                <td style="border: 1px solid #ddd; padding: 8px;">The plugin is functional and being tested, but some features may still be refined.</td>
```
```suggestion
                <td style="border: 1px solid #ddd; padding: 8px;">The plugin is functional and being tested but some features may still be refined.</td>
```

edan-bainglass reviewed (changes requested):
Thanks @superstar54. I suggested a few changes. 

Also, a few issues with the available plugins list:

```
Bader charge analysis (aiida-bader) ☐ -> why not aiidalab-bader?
Phonons and IR/Raman (aiidalab-qe-vibroscopy) ☐ -> icon not working
Muon spectroscopy (aiidalab-qe-muon) ☐
Core-level spectroscopy (aiida-qe-xspec) ✅ -> again, why not aiidalab-...?
Wannier functions (aiidalab-qe-wannier90) ☐
Habbard parameter (aiidalab-qe-hp) ☐ -> Hubbard, not Habbard
```

superstar54 commented:
@edan-bainglass Thanks for the review. I implemented the review suggestions.

> Phonons and IR/Raman (aiidalab-qe-vibroscopy) ☐ -> icon not working

I removed the checkbox `icon` if the plugin is not installed.
